From faf31108e10cf103ae015c962ccac2253d2ef8cb Mon Sep 17 00:00:00 2001
From: Antonio Francisco <ajoaoff@gmail.com>
Date: Thu, 27 May 2021 09:07:01 -0300
Subject: [PATCH 1/2] Method to allow new Action classes to be set

Added a method to support new action classes
implemented in other modules.
---
 v0x04/flow.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/v0x04/flow.py b/v0x04/flow.py
index 5e27ad8..524dbdd 100644
--- a/v0x04/flow.py
+++ b/v0x04/flow.py
@@ -178,6 +178,10 @@ class Action(ActionFactoryBase):
         OFActionSetQueue: ActionSetQueue
     }
 
+    @classmethod
+    def add_action_class(cls, class_name, new_class):
+        cls._action_class[class_name] = new_class
+
 
 class Flow(FlowBase):
     """High-level flow representation for OpenFlow 1.0.

From 9f4fc6b8580cc6c397964e370eb333b3c98d576c Mon Sep 17 00:00:00 2001
From: Antonio Francisco <ajoaoff@gmail.com>
Date: Wed, 30 Jun 2021 17:37:20 -0300
Subject: [PATCH 2/2] Add comment to method `add_action_class`

---
 v0x04/flow.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/v0x04/flow.py b/v0x04/flow.py
index 524dbdd..93f2f39 100644
--- a/v0x04/flow.py
+++ b/v0x04/flow.py
@@ -180,6 +180,10 @@ class Action(ActionFactoryBase):
 
     @classmethod
     def add_action_class(cls, class_name, new_class):
+        """Add a new action.
+
+        To be used by modules implementing Experimenter Actions.
+        """
         cls._action_class[class_name] = new_class
 
 
